pkg/envoy/envoyadminclient.go

```diff
@@ -120,3 +130,43 @@ func (a *EnvoyAdminClient) GetEnvoyVersion() (string, error) {
 
 	return fmt.Sprintf("%s", version), nil
 }
+
+func (a *EnvoyAdminClient) GetConfigDump(resourceType string, resourceName string) (string, error) {
```

Comment:
Reusing the `EnvoyAdminClient` (and pulling in the dependent packages) adds additional 10MB to the `cilium-dbg` binary.

```
#main
root@kind-worker:/home/cilium# ls -lah /usr/bin/cilium-dbg
-rwxr-xr-x 1 root root    73M Aug 16 06:39 cilium-dbg

#feature-branch
root@kind-worker:/home/cilium# ls -lah /usr/bin/cilium-dbg
-rwxr-xr-x 1 root root    87M Aug 16 06:24 cilium-dbg
```

Will either try to move this into a subpackage or introduce a copy in the `cilium-dbg` package.

Reply:
The latest update introduces a separate version of the  `EnvoyAdminClient` in `cilium-dbg`. This way we're back at 72M file size :pray: 

There are also other reasons to separate the client between `cilium-agent` & `cilium-dbg`:

* Only exposing the required API for their use-cases,
* Avoid context issues (e.g. if cilium-dbg would start using setting the loglevel but lacks having the debuglevel configured (because this is only configured on the agent)

/cc @asauber (`cilium/cli`)

Reply:
Hmm, good to see. This makes me think that we should add a workflow which checks for unexpected binary or container size increases across the board.